### PR TITLE
FIREFLY-994: Create chart view that support multiple charts from different tables

### PR DESCRIPTION
**Pull request by loitly**
Ticket: https://jira.ipac.caltech.edu/browse/FIREFLY-994
Test: https://fireflydev.ipac.caltech.edu/firefly-994-multi-table-chart-view/firefly/

Basic description of the added features, copied from ticket.
When useChartWorkArea is enabled,
- Chart Viewer may contains Active Charts and Pinned Charts
  - Displayed as tabs or side-by-side
  - Top toolbar added with help anchor
- Pin Chart button is available with Active Chart is visible
- When pinning chart
  - Ask for title if one does not exist
  - Default to table's title if exists otherwise blank
  - Not allowed after max number of pinned charts is reached (set at 6)
  - On first pin, switch to side-by-side view so the user can visually see a new chart is added to Pinned Charts tab
- When a table is deleted or removed, remove all pinned charts associated with the table.
- Show Table button: set table of the highlighted chart to active tab

Additional changes made to PR not mentioned in ticket:
- convert ChartPanel to functional component
- fix: fail to send email from docker container due to unsupported protocol
- fix: chart toolbar composition and layout

### CODE REVIEW

**Review by ejoliet — CHANGES_REQUESTED** [review]
This PR has all the pieces we need to move forward with the chart comparison and spectra viewer. 
Couple of issues i found during my testing that i think would be good to look at before merging:

- bring 2 catalogs or more, the coverage map has few weird behavior:
  - it becomes 'closable' - is that a behavior that was changed for a reason? I think it needs to stay and nt enable to remove/close.
  - if i start with one catalog, the coverage is "FITS" , then if i add one search more, it becomes "HiPS" and a third one makes it "Aitoff" , there is something going on - those coverage should stay with same type (fits or hips), like in ops. 

- If i "pin chart", click on 'as tabs' and then go to 'pinned chart' and close the pinned chart, the "pin chart" and other buttons have disappeared and they don't come back even if i do another search, they should stay

Look at those and let me know if you cannot reproduce.

The rest looks fine and is pretty nice to use so that's great!
Thanks!

**Comment by loitly** [comment]
> * it becomes 'closable' - is that a behavior that was changed for a reason? I think it needs to stay and nt enable to remove/close.

I can't duplicate.

> * if i start with one catalog, the coverage is "FITS" , then if i add one search more, it becomes "HiPS" and a third one makes it "Aitoff" , there is something going on - those coverage should stay with same type (fits or hips), like in ops.

I see the same switching behavior in nightly.

**Comment by ejoliet** [comment]
> > * it becomes 'closable' - is that a behavior that was changed for a reason? I think it needs to stay and nt enable to remove/close.
> 
> I can't duplicate.
> 
> > * if i start with one catalog, the coverage is "FITS" , then if i add one search more, it becomes "HiPS" and a third one makes it "Aitoff" , there is something going on - those coverage should stay with same type (fits or hips), like in ops.
> 
> I see the same switching behavior in nightly.

I can show you how to duplicate, it's difficult to write it down.

File: src/firefly/js/charts/ChartsCntlr.js [hunk 1]
```diff
@@ -681,11 +681,11 @@ function reduceData(state={}, action={}) {
         case (CHART_MOUNTED) :
         {
             const {chartId} = action.payload;
-            if (has(state, chartId)) {
+            // if (has(state, chartId)) {
```
Comment (robyww) [inline src/firefly/js/charts/ChartsCntlr.js:684]: probably should just remove this commented code.

**Review by robyww — APPROVED** [review]
Looks Good!

File: src/firefly/html/firefly.html [hunk 2]
```diff
@@ -19,6 +19,9 @@
                 hips: {
                     readoutShowsPixel: true
                 },
+                charts: {
+                    useChartWorkArea: true
+                },
```
[the remark this answers is not in the document]
Comment (loitly) [reply src/firefly/html/firefly.html:24]: I'll have to enable it at the `html` level.  We want this feature in the app and not the api, by default.   Does this make sense?

File: src/firefly/html/css/global.css [hunk 3]
```diff
@@ -107,6 +107,19 @@ div.rootStyle img {
...
+    border: 1px solid #acacac;
+    border-radius: 2px;
+    white-space: nowrap;
+}
+
+.button.text:hover {
+    border: 1px solid #878787;
+    background-color: #c4c4c4;
+}
+
```
[the remark this answers is not in the document]
Comment (loitly) [reply src/firefly/html/css/global.css:122]: No.  Only applies to `div` with class='button text'.

**Pull request merged**